wandb/sdk/internal/file_stream.py

```diff
@@ -249,43 +298,65 @@ class FileStreamApi:
     TODO: Differentiate between binary/text encoding.
     """
 
-    Finish = collections.namedtuple("Finish", ("exitcode"))
-    Preempting = collections.namedtuple("Preempting", ())
-    PushSuccess = collections.namedtuple("PushSuccess", ("artifact_id", "save_name"))
+    class Finish(NamedTuple):
+        exitcode: int
+
+    class Preempting(NamedTuple):
+        pass
+
+    class PushSuccess(NamedTuple):
+        artifact_id: str
+        save_name: str
 
     HTTP_TIMEOUT = env.get_http_timeout(10)
     MAX_ITEMS_PER_PUSH = 10000
 
-    def __init__(self, api, run_id, start_time, settings=None):
-        if settings is None:
-            settings = dict()
+    def __init__(
+        self,
+        api: "internal_api.Api",
+        run_id: str,
+        start_time: float,
+        settings: Optional[
+            Union[
+                "wandb.sdk.wandb_settings.Settings",
+                "settings_static.SettingsStatic",
+                dict,
+            ]
+        ] = None,
+    ) -> None:
+        settings = settings or dict()
         # NOTE: exc_info is set in thread_except_body context and readable by calling threads
-        self._exc_info = None
+        self._exc_info: Optional[
+            Union[
+                Tuple[Type[BaseException], BaseException, TracebackType],
+                Tuple[None, None, None],
+            ]
+        ] = None
         self._settings = settings
         self._api = api
         self._run_id = run_id
         self._start_time = start_time
         self._client = requests.Session()
-        self._client.auth = ("api", api.api_key)
-        self._client.timeout = self.HTTP_TIMEOUT
+        self._client.post = partial(self._client.post, timeout=self.HTTP_TIMEOUT)  # type: ignore
```

Comment:
@wandb/sdk-team does this look like a safe change to you?

Reply:
given your other comments looks like the previous logic didn't have timeout

Reply:
I think this is the right move for now, I think we are close to doing more network testing so we can do this proper when we are better prepared to test it.

I tracked down the original PR:
https://github.com/wandb/client/commit/509025c14c791e4e42704ccf777f10e240222360#diff-9e27f3fb488f65e828f371a3efc7c61d87d96284f23fe3c4242c196258a8fe1eR81